src/librustc/session/mod.rs

```diff
@@ -634,6 +639,10 @@ impl Session {
     /// Returns the panic strategy for this compile session. If the user explicitly selected one
     /// using '-C panic', use that, otherwise use the panic strategy defined by the target.
     pub fn panic_strategy(&self) -> PanicStrategy {
+        if *self.force_panic_abort.get() {
```

Comment:
could this also be `cfg!(miri)`?

Reply:
This is useful for anyone using xargo I think, not just miri.

Reply:
As @bjorn3 said, this has the benefit of making things more consistent in general. The interaction of `PanicStrategy`, `-C panic`, and `libpanic_abort` is already pretty confusing, so I think it would be a mistake to make the logic even more complication by introducing a `cfg`.